pom.xml

```diff
@@ -139,13 +139,23 @@
             <modules>
                 <module>tests-spark310+</module>
             </modules>
+            <repositories>
+                <!--
+                TODO remove once Apache Spark 3.1.1 is released.
+                Last 3.1.1-SNAPSHOT is before SPARK-34223
+                -->
+                <repository>
+                    <id>spark311rc</id>
+                    <url>https://repository.apache.org/content/repositories/orgapachespark-1365/</url>
```

Comment:
I'm not thrilled putting this in.  If this is expected to be very temporary, why not just hold off merging this PR until 3.1.1 is officially released or the snapshot build is fixed?

Reply:
ok, I we can make it a property or environment variable passed from outside, which is actually convenient now that we are waiting for rc3 for 3.1.1
e.g.  this would fail
```
$ find ~/.m2/repository -path \*/spark\*/3.1.1 -type d | xargs rm -r
$ mvn -Pspark311tests,snapshot-shims test -Dspark.rc.repo=https://repository.apache.org/content/repositories/orgapachespark-1364 -Dcuda.version=cuda11 -DwildcardSuites=com.nvidia.spark.rapids.InsertPartition311Suite
```

this would succeed
```
$ find ~/.m2/repository -path \*/spark\*/3.1.1 -type d | xargs rm -r
$ mvn -Pspark311tests,snapshot-shims test -Dspark.rc.repo=https://repository.apache.org/content/repositories/orgapachespark-1365 -Dcuda.version=cuda11 -DwildcardSuites=com.nvidia.spark.rapids.InsertPartition311Suite
```

Reply:
But then we'd have to update all the CI/CD scripts or they will all break.

I think we should just wait until 3.1.1 gets sorted out before merging this change.  We can't say 0.4 supports 3.1 until 3.1 is out anyway. :smile:  In the end I think it's a lot less code churn and less risk.